src/Text/Pandoc/Readers/LaTeX.hs

```diff
@@ -728,16 +728,19 @@ dolstinline = do
 doLHSverb :: PandocMonad m => LP m Inlines
 doLHSverb = codeWith ("",["haskell"],[]) <$> manyTill (satisfy (/='\n')) (char '|')
 
+-- converts e.g. \SI{1}[\$]{} to "$ 1" or \SI{1}{\euro} to "1 €"
 dosiunitx :: PandocMonad m => LP m Inlines
 dosiunitx = do
   skipopts
   value <- char '{' >> (mconcat <$> manyTill inline (char '}'))
```

Comment:
Try

    value <- tok

This will handle escapes, etc., better than what you have.